pyro/distributions/hmm.py

```diff
@@ -405,6 +465,38 @@ def filter(self, value):
         return MultivariateNormal(loc, precision_matrix=precision,
                                   validate_args=self._validate_args)
 
+    def posterior(self, likelihood):
+        """
+        Create a new :class:`GaussianHMM` incorporating likelihood information.
+
+        This is useful for generating posterior samples using :meth:`rsample` .
+
+        :param likelihood: A distribution representing likelihoods of
+            observations, or equivalently, noisy observations.
+        :type likelihood: ~torch.distributions.MultivariateNormal or
+            ~torch.distributions.Independent of ~torch.distributions.Normal
+        :return: A new instance updated with the likelihood.
+        :rtype: GaussianHMM
+        """
+        assert (isinstance(likelihood, torch.distributions.MultivariateNormal) or
+                (isinstance(likelihood, torch.distributions.Independent) and
+                 isinstance(likelihood.base_dist, torch.distributions.Normal)))
+        new = self._get_checked_instance(GaussianHMM)
+        new.hidden_dim = self.hidden_dim
+        new.obs_dim = self.obs_dim
+        new._trans = self._trans
+        new._obs = self._obs + mvn_to_gaussian(likelihood).event_pad(left=self.hidden_dim)
```

Comment:
Hmm at this point the distribution is no longer normalized, so `.log_prob()` will be invalid. I think I'll change to compute the log normalizer here. We may even be able to return that for use in inference.

Reply:
I've fixed and updated to use the new `.conjugate_update()` interface of #2289 .